### ISSUE TITLE
Support fractional numbers in window.zoomlevel from keyboard zoom actions

### ISSUE THREAD

**Issue by mosfetish** [microsoft/vscode#120516]
When using ctrl++, ctrl+-, and ctrl+0 to zoom/shrink/reset window zoom, window.zoomlevel in settings.json is updated or completely removed, as in the case of resetting with ctrl+0.

A setting applied in settings.json should not be updated/removed by using simple keyboard actions.

Many users set window.zoomlevel to something like 0.4 to increase the size of the window chrome since there is no other way to do it in Code. But they still need to use the keyboard zoom from time to time.

So here's what happens with window.zoomlevel in settings.json when using the keyboard zoom actions:

```
window.zoomlevel: 0.4
ctrl++:
window.zoomlevel: 1
ctrl+-
[window.zoomlevel is removed from settings.json]
ctrl++
window.zoomlevel: 1
```

What should happen if Code is respecting the user setting is this:

```
window.zoomlevel: 0.4
ctrl++:
window.zoomlevel: 1
ctrl+-
window.zoomlevel: 0.4
ctrl++
window.zoomlevel: 1

```

In other words, the user setting of 0.4 should become another step in the keyboard zoom sequence and not be removed. And using ctrl+0 when a user-defined setting is in settings.json should reset the zoom to the user-defined setting, and not remove the setting.

It's pretty frustrating to do temporary zoom-in and have the setting overwritten or deleted.
